account_invoice_import/wizard/account_invoice_import.py

```diff
@@ -209,13 +211,45 @@ def _prepare_create_invoice_vals(self, parsed_inv, import_config):
         currency = self._match_currency(
             parsed_inv.get("currency"), parsed_inv["chatter_msg"]
         )
-        journal_id = (
-            amo.with_context(
-                default_move_type=parsed_inv["type"], company_id=company.id
+        if parsed_inv["type"] in ("in_invoice", "in_refund") and import_config.get(
+            "journal"
+        ):
+            journal_id = import_config["journal"].id
+        elif parsed_inv.get("journal"):
```

Comment:
@alexis-via can you please use dedicated methods for these blocks?

Reply:
We are already in a prepare method, so I think this small piece of code is at its right place.